irohad/ordering/impl/ordering_gate_impl.cpp

```diff
@@ -53,15 +47,16 @@ namespace iroha {
       }
       log_->info("transactions in proposal: {}", transactions.size());
 
-      model::Proposal proposal(transactions);
-      proposal.height = request->height();
-      handleProposal(std::move(proposal));
+      auto proposal = transport_->getProposal(request);
 
+      proposal->height = request->height();
+      transport_->subscriber_->handleProposal(proposal);
+      //handleProposal(proposal);
       return grpc::Status::OK;
     }
 
-    void OrderingGateImpl::handleProposal(model::Proposal &&proposal) {
-      proposals_.get_subscriber().on_next(proposal);
+    void OrderingGateImpl::handleProposal(std::shared_ptr<model::Proposal> proposal) {
```

Comment:
This proposal is only used in this method, so why use `shared_ptr`?

Reply:
I talked with Fyodor and we decidet to use shared_ptr in order to be safe from seg faults, if client destructs object in another thread